include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -227,4 +231,189 @@ std::string get_cigar_string(alignment_type && alignment,
+        cigar_op = *std::ranges::begin(cigar_view);
+        std::ranges::next(std::ranges::begin(cigar_view));
+    }
+
+    if (std::from_chars(buffer.begin(), buffer_end, cigar_count).ec != std::errc{})
+        throw std::runtime_error{"Corrupted cigar string encountered"};
+
+    if (is_char<'S'>(cigar_op)) // check for soft clipping at the beginning
+    {
+        sc_begin_count = cigar_count;
+    }
+    else
+    {
+        update_lengths_fn();
+        result.push_back({cigar_op, cigar_count});
+    }
+
+    // parse the rest of the cigar
+    // -----------------------------------------------------------------------------------------------------------------
+    while (std::ranges::begin(cigar_view) != std::ranges::end(cigar_view)) // until stream is not empty
+    {
+        // write next count into buffer and keep track of ints length in variable idx
+        buffer_end = (std::ranges::copy(cigar_view | view::take_until_or_throw(is_cigar_op), buffer.data())).out;
+        cigar_op = *std::ranges::begin(cigar_view);
+        std::ranges::next(std::ranges::begin(cigar_view));
+
+        if (std::from_chars(buffer.begin(), buffer_end, cigar_count).ec != std::errc{})
+            throw std::runtime_error{"Corrupted cigar string encountered"};
+
+        if (is_char<'S'>(cigar_op)) // we are at the end, hard clipping afterwards can be ignored
+        {
+            sc_end_count = cigar_count;
+            return {result, ref_length, seq_length};
+        }
+        update_lengths_fn();
+        result.push_back({cigar_op, cigar_count});
+    }
+    return {result, ref_length, seq_length};
+}
+
+/*!\brief Transforms a std::vector of operation-count pairs (representing the cigar string).
+ * \ingroup alignment_file
+ *
+ * \tparam alignment_type The type of alignment; Must model seqan3::tuple_like_concept and all tuple element types
+ *                        must model seqan3::aligned_sequence_concept.
+ *
+ * \param[out] alignment  The alignment to fill with gaps according to the cigar information.
+ * \param[in]  cigar      The cigar information given as a std::vector of operation-count pairs.
+ *
+ * \details
+ *
+ * ### Example:
+ *
+ * Given the following cigar string "4M2I5M2D1M", the cigar information extracted by seqan3::detail::parse_cigar
+ * would be "[(M,4), (I,2), (M,5), (D,2), (M,1)]". Given those cigar information, and an alignment variable containing
+ * the two unaligned sequences "(ATGGCGTAGAGC, ATGCCCCGTTGC)", the alignment will be filled with the following gaps:
+ *
+ * ```
+ * ATGG--CGTAGAGC
+ * ||||  |||||  |
+ * ATGCCCCGTTG--C
+ * ```
+ */
+template<tuple_like_concept alignment_type>
+//!\cond
+    requires std::tuple_size_v<remove_cvref_t<alignment_type>> == 2 &&
+             detail::all_satisfy_aligned_seq<detail::tuple_type_list_t<alignment_type>>
+//!\endcond
+void alignment_from_cigar(alignment_type & alignment, std::vector<std::pair<char, size_t>> const & cigar)
```

Comment:
hmm I don't know if everyone would know what I mean by operations?

Reply:
why not just `cigarstring_to_alignment` and `alignment_to_cigarstring`?

Reply:
because I have one function in between.
cigarstring_to_extracted_information (operations, soft clipping, lengths,...)
part_of_extracted_information_to_alignment